packages/sdk/src/sdk/services/Solana/programs/ClaimableTokensClient/ClaimableTokensClient.ts

```diff
@@ -113,38 +127,60 @@ export class ClaimableTokensClient {
     const { mint, token } = parseMintToken(args.mint, this.mints)
     const feePayer = feePayerOverride ?? (await this.client.getFeePayer())
     const userBank = await this.deriveUserBank(args)
-    const userBankAccount = await this.client.connection.getAccountInfo(
-      userBank
-    )
-    if (!userBankAccount) {
-      this.logger.debug(`User bank ${userBank} does not exist. Creating...`)
-      const createUserBankInstruction =
-        ClaimableTokensProgram.createAccountInstruction({
-          ethAddress: ethWallet,
-          payer: feePayer,
-          mint,
-          authority: this.authorities[token],
-          userBank,
-          programId: this.programId
-        })
-      const confirmationStrategyArgs =
-        await this.client.connection.getLatestBlockhash()
-      const message = new TransactionMessage({
-        payerKey: feePayer,
-        recentBlockhash: confirmationStrategyArgs.blockhash,
-        instructions: [createUserBankInstruction]
-      }).compileToLegacyMessage()
-      const transaction = new VersionedTransaction(message)
-      const signature = await this.sendTransaction(transaction)
-      const confirmationStrategy = { ...confirmationStrategyArgs, signature }
-      await this.client.connection.confirmTransaction(
-        confirmationStrategy,
-        'finalized'
+
+    // If already fetching/creating this user bank, wait for the existing promise.
+    // Then assume the user bank has been created and return with didExist=true.
+    // Don't attempt to catch and retry or concurrency will again be introduced.
+    if (this._pendingUserBankCreationPromises[userBank.toBase58()]) {
+      await this._pendingUserBankCreationPromises[userBank.toBase58()]
+      return { userBank, didExist: true }
```

Comment:
Why not just return the existing promise if it exists? Maybe I'm missing something here. Also curious about assuming `didExist`? Is the idea that if we find a promise already, then the second call wouldn't technically be the one creating the user bank, so it "existed" already even if we're waiting on the same promise?

Separately I'm also wondering why we even have `didExist`.

Reply:
Yes! I felt having multiple calls claim to have "created" the user bank would be confusing, so made any subscribers override that to set `didExist` to true as if to say "oh i just used what the other guy did, I didn't make it"

and yeah, the reason we had `didExist` is mostly for metrics/logging/debugging things